### PR TITLE
feat(platform): message popover

### LINKED ISSUE #8329: Message Handling for Forms with Message Popover
Feature request

We deal with multiple forms across our solution. SAPUI5 provides the message popover for handling messages on such forms.
https://sapui5.netweaver.ondemand.com/sdk/#/entity/sap.m.MessagePopover

We do not see such a component being provided my the fundamental-ngx team and kindly request this as a centrally provided feature.
image

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #8329

## Description
Introduced new component for the platform library: Message popover.
Added documentation examples. refactored form controls in platform library

## Screenshots
image
image
image

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/i18n/src/lib/languages/georgian.ts [hunk 1]
```diff
@@ -538,6 +538,19 @@ export const FD_LANGUAGE_GEORGIAN: FdLanguage = {
...
+            email: 'Email is invalid',
+            max: 'The field exceeds maximum value',
+            maxLength: 'The field exceeds maximum length',
+            min: 'The field value is less than allowed',
+            minLength: 'The field length is less than allowed',
+            pattern: 'The field value is invalid',
+            required: 'The field is mandatory',
+            requiredTrue: 'The field is mandatory'
+        }
+    },
```
Comment (g-cheishvili) [inline libs/i18n/src/lib/languages/georgian.ts:553]:
```suggestion
    platformMessagePopover: {
        allErrors: 'ყველა',
        defaultErrors: {
            email: 'ელ-ფოსტა არასწორია',
            max: 'ველის მნიშვნელობა აღემატება დაშვებულ მაქსიმუმს',
            maxLength: 'ველის მოცულობა აღემატება დაშვებულ მაქსიმუმს',
            min: 'ველის მნიშვნელობა აუცილებელ მინიმუმზე ნაკლებია',
            minLength: 'ველის მოცულობა აუცილებელ მინიმუმზე ნაკლებია',
            pattern: 'ველის მნიშვნელობა არასწორია',
            required: 'ველი სავალდებულოა',
            requiredTrue: 'ველის შეყვანა სავალდებულოა'
        }
    },
```

File: libs/platform/src/lib/form/input-message-group-with-template/input-message-group-with-template.component.html [hunk 2]
```diff
@@ -13,6 +13,7 @@
         <ng-container *ngTemplateOutlet="triggerItemTemplate"></ng-container>
     </fd-popover-control>
     <fd-popover-body>
-        <ng-content select="fd-form-message"></ng-content>
+        <ng-content></ng-content>
+<!--        <ng-content select="fd-form-message"></ng-content>-->
```
Comment (platon-rov) [inline libs/platform/src/lib/form/input-message-group-with-template/input-message-group-with-template.component.html:17]: forgot to remove?

**Review by platon-rov — COMMENTED** [review]
good job! Left minor things

**Comment by InnaAtanasova** [comment]
1. Missing focus trap on the component. When you open the Message Popover and press Tab the focus jumps out of the component. 
When the Popover is closed the focus should go to the trigger.

**Comment by InnaAtanasova** [comment]
More a visual thing but the item should not have rounded borders:

Screenshot 2022-11-03 at 4 08 11 PM

**Comment by InnaAtanasova** [comment]
Double scroll when the popover body appears under the control:
Screenshot 2022-11-03 at 4 12 11 PM

**Comment by InnaAtanasova** [comment]
I might be wrong but there's no example showing error details. Here's an example from UI5:

Screenshot 2022-11-03 at 4 13 16 PM

Clicking on the item opens a second screen with description:
Screenshot 2022-11-03 at 4 13 23 PM

**Comment by InnaAtanasova** [comment]
Could you provide example with different types of messages so we can test the filtering option?
Screenshot 2022-11-03 at 4 16 10 PM

**Comment by N1XUS** [comment]
@InnaAtanasova for different types of errors, you can try entering weak password in form generator example. Meanwhile I'll add some new types of errors to the examples

File: libs/docs/platform/message-popover/message-popover-docs.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ];
+
+    messagePopoverContainerExample: ExampleFile[] = [
+        getExampleFile('form-container-example/form-container-example.component.html'),
+        getExampleFile('form-container-example/form-container-example.component.ts')
+    ];
+
+    messagePopoverGeneratorExample: ExampleFile[] = [
+        getExampleFile('form-generator-example/form-generator-component-example.component.html'),
+        getExampleFile('form-generator-example/form-generator-component-example.component.ts')
```
Comment (g-cheishvili) [inline libs/docs/platform/message-popover/message-popover-docs.component.ts:21]:
```suggestion
        getExampleFile('form-generator/form-generator-component-example.component.html'),
        getExampleFile('form-generator/form-generator-component-example.component.ts')
```

**Comment by InnaAtanasova** [comment]
Looking great! Impressive work! Thanks for addressing my comments. 
One minor things:
when details screen is open the back button should be focussed:
Screenshot 2022-11-04 at 11 39 32 AM

**Review by InnaAtanasova — APPROVED** [review]

File: libs/docs/platform/message-popover/message-popover-docs.component.html [hunk 4]
```diff
@@ -0,0 +1,81 @@
+<fd-docs-section-title id="default" componentName="message-popover">
+    Message Popover with Simple forms
+</fd-docs-section-title>
+<description>
+    <p>Message popover component supports both template-driven and reactive forms.</p>
+    <p>
+        To define human-readable form item label text, developers need to use
```
Comment (platon-rov) [inline libs/docs/platform/message-popover/message-popover-docs.component.html:7]:
```suggestion
        To define human-readable error message text, developers need to use
```

File: libs/docs/platform/message-popover/message-popover-docs.component.html [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        <code>fdpMessagePopoverFormItem</code> directive.
```
Comment (platon-rov) [inline libs/docs/platform/message-popover/message-popover-docs.component.html:8]: Does it mean that we cannot specify error type and heading/descriptionwith this approach? Is it intended?

**Review by g-cheishvili — APPROVED** [review]

File: libs/docs/platform/message-popover/message-popover-docs.component.html (hunk 4 above)
Comment (N1XUS) [reply libs/docs/platform/message-popover/message-popover-docs.component.html:7]: Hey @platon-rov. The original text is actually correct. With fdpMessagePopoverFormItem developers can define field name if they are not using form field component from platform lib (as example, they use core's controls).

File: libs/docs/platform/message-popover/message-popover-docs.component.html (hunk 5 above)
Comment (N1XUS) [reply libs/docs/platform/message-popover/message-popover-docs.component.html:8]: @platon-rov developers can define it on module level with `withConfig` method, or provide it on component level with `FDP_MESSAGE_POPOVER_CONFIG ` token. We support 2 options: plain string (will be threatened as a heading only), and more complex object which contains header text, description text and error type

**Review by platon-rov — APPROVED** [review]

**Pull request merged**